modules/extensions/src/data-filter/data-filter-extension.ts

```diff
@@ -43,6 +45,11 @@ export type DataFilterExtensionProps<DataT = any> = {
    * Returns either a number (if `filterSize: 1`) or an array of numbers.
    */
   getFilterValue?: Accessor<DataT, number | number[]>;
+  /**
+   * Accessor to retrieve the category for each object that it will be filtered by.
+   * Returns either a number (if `filterSize: 1`) or an array of numbers.
+   */
+  getFilterCategory?: Accessor<DataT, number | number[]>;
```

Comment:
Would this work with a typed array passed into `data.attributes.getFilterCategory` as well?